pkg/minikube/cruntime/containerd.go

```diff
@@ -277,6 +279,109 @@ func (r *Containerd) RemoveImage(name string) error {
 	return removeCRIImage(r.Runner, name)
 }
 
+func gitClone(cr CommandRunner, src string) (string, error) {
+	// clone to a temporary directory
+	rr, err := cr.RunCmd(exec.Command("mktemp", "-d"))
+	if err != nil {
+		return "", err
+	}
+	tmp := strings.TrimSpace(rr.Stdout.String())
+	cmd := exec.Command("git", "clone", src, tmp)
+	if _, err := cr.RunCmd(cmd); err != nil {
+		return "", err
+	}
+	return tmp, nil
+}
+
+func downloadRemote(cr CommandRunner, src string) (string, error) {
```

Comment:
add unit test for this ? or maybe break down the logic of detecting remote and add unit tests, so we know exactly what user would input

Reply:
Right, but then we need to start a web server or something to host the files to build (the ones in testdata/build) ?

Reply:
I think the detection part of the (the scheme detection) could be put in a separate func with unit test ?

Reply:
We only implement it for containerd, both docker and podman have their own (internal) implementation...